packages/vats/src/proposals/upgrade-mintHolder-atom-proposal.js

```diff
@@ -0,0 +1,56 @@
+import { makeTracer } from '@agoric/internal';
+import { E } from '@endo/far';
+
+const trace = makeTracer('upgrade mintHolder ATOM', true);
+
+export const upgradeMintHolder = async (
+  {
+    consume: {
+      contractKits: contractKitsP,
+      instancePrivateArgs: instancePrivateArgsP,
+    },
+  },
+  options,
+) => {
+  trace('Start contract upgrade');
+
+  const { contractRef } = options.options;
+  assert(contractRef.bundleID, 'mintHolder bundleID not found');
+
+  const [contractKits, instancePrivateArgs] = await Promise.all([
+    contractKitsP,
+    instancePrivateArgsP,
+  ]);
+
+  const mintHolderKit = Array.from(contractKits.values()).filter(
+    kit => kit.label && kit.label.match(/ATOM/),
```

Comment:
lots of vats have `ATOM` in their names. Let's find a better way to uniquely identify the mintHolder vat.

The name also has 'mintHolder' in it, so that's not unique yet. On MainNet, there are vats for ATOM, stATOM, and stkATOM as well as two different kinds of USDC, USDC, and DAI.  should we specify the vat number, which is unique (though different) on each platform?

I think I'll post something to the kernel team. They often have opinions and ideas about this kind of thing.

Reply:
I think everyone would be satisfied if we first filter for the installation, and then look for the exact name. At the moment, there's only one installation for mintHolder, so we can just get the unique mintHolder installation from the promise space.

Reply:
Please clarify if you meant `instance` instead of `installation`?
I am asking because, if I am not mistaken, the contractKits only hold the instance, adminFacet, creatorFacet and publicFacet.

Reply:
I did mean `installation`, but you're right, that's not available.  I was happy with that because one `installation` is shared across many instances. The instances aren't helpful for this.

The vat names are unique, but the `label`s used in `contractKits` are just the [issuer name](https://github.com/Agoric/agoric-sdk/blob/a66f6a1408016746f4f480241f3d6efa53732841/packages/inter-protocol/src/proposals/addAssetToVault.js#L109).  

Dan had suggested using `__getMethodNames__()` on the publicFacet to distinguish `mintHolders`. The publicFacet of a `mintHolder` is the `issuer` itself.  (I didn't realize you could call that on a remote reference, but Dan thinks so.)   If you use an exact match on the label, and verify that the publicFacet has a `makeEmptyPurse` method, then I'll be comfortable that you've found the right contract and it's a mintHolder. Does that work?  Is the issuer name used for the label sufficient to distinguish the varieties of *USDT* and *DAI*?

Reply:
The tests showed that using strict equality operator to compare the label with the issuer name, is indeed capable of distinguish the varieties of USDT, USDC and DAI assets.

I was not able to find an example of __getMethodNames__() being used on Agoric code, and I while trying different implementations for it, I was not successful to find one that would return a list of exposed methods by the mintHolder publicFacet.

Although, at commit [b9c5acb](https://github.com/Agoric/agoric-sdk/pull/10617/commits/b9c5acb5341ff6b909eadaf4214267c68b50d9f6), I followed the approach you suggested by calling the `makeEmptyPurse` method and verify that the core-eval did not throw during the test execution.

Is this sufficient to confirm that we are interacting with the expected contract?